### ISSUE TITLE
Mixpanel BufferedConsumer throws `MixpanelException: Mixpanel error: a temporary failure occurred`

### ISSUE THREAD

**Issue by athul-mindtickle** [mixpanel/mixpanel-python#80]
I use `python:3.7.3-stretch` docker image. It is run in AWS EKS. 
The mixpanel library version is `mixpanel==4.5.0`.

I use the `BufferedConsumer`.

```python
from mixpanel import Mixpanel, BufferedConsumer
mxp_consumer = BufferedConsumer(max_size=BaseConfig.BATCH_SIZE*5)
mp = Mixpanel(BaseConfig.TOKEN, consumer=mxp_consumer)
```

I flush the endpoint once it reaches either the time threshold or the batch size threshold. The corresponding code snippet is given below. 

```python
if (((len(events) >= BaseConfig.BATCH_SIZE) or
     (datetime.now() > (start + timedelta(minutes=BaseConfig.BATCH_FLUSH_TIME)))) and
 (len(events) > 0)):

	try:
		await process_and_send_to_mixpanel(events, mp) 
		mxp_consumer.flush() # Flush the buffered consumer
		events.clear()  # Clear list
		start = datetime.now()  # Reset Time

	except Exception as e:
		logger.error(e, exc_info=True)
		logger.error(e)
		
```

inside the `process_and_send_to_mixpanel(events, mp)` function I make calls to both `/engage` `/track` endpoints using the below functions.

```python
mp.people_set(str(distinct_id),
              people_properties,
              meta={"$time": datetime.now().isoformat()})
		

mp.track(str(distinct_id),
         str(mxp_event_event_name),
         {**mxp_event,
          "time": datetime.now().isoformat()})

```

This code works without errors for 95% of the time. 

but I get the **BELOW GIVEN ERROR**

```python
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/mixpanel/__init__.py", line 598, in _flush_endpoint
    self._consumer.send(endpoint, batch_json, self._api_key)
  File "/usr/local/lib/python3.7/site-packages/mixpanel/__init__.py", line 488, in send
    self._write_request(self._endpoints[endpoint], json_message, api_key)
  File "/usr/local/lib/python3.7/site-packages/mixpanel/__init__.py", line 519, in _write_request
    raise MixpanelException('Mixpanel error: {0}'.format(response['error']))
mixpanel.MixpanelException: Mixpanel error: a temporary failure occurred

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
    mxp_consumer.flush()
  File "/usr/local/lib/python3.7/site-packages/mixpanel/__init__.py", line 590, in flush
    self._flush_endpoint(endpoint)
  File "/usr/local/lib/python3.7/site-packages/mixpanel/__init__.py", line 603, in _flush_endpoint
    six.raise_from(mp_e, orig_e)
  File "<string>", line 3, in raise_from
mixpanel.MixpanelException: Mixpanel error: a temporary failure occurred
```

The response from the server is `{'error': 'a temporary failure occurred', 'status': 0}`.

I kill the process and re run again. There are no issues. But again, at random, this error arises.

**Comment by seizethedave** [comment]
Thanks for the report. The problem you describe will probably be addressed in this library by adding Mixpanel deduplication ID properties (`$insert_id`) as described in the [HTTP spec](https://developer.mixpanel.com/docs/http#event-deduplication), and retrying failures such as the one you encountered. Until that has been added to mixpanel-python, you could work around this by:

- forming & passing along your own $insert_id, so that retrying can be safely done.
- handling and ignoring MixpanelException when you call `people_set` or `track` (assuming you finalize your session by calling `BufferedConsumer.flush`)
- catching MixpanelException when you call `BufferedConsumer.flush`, and trying it again up to a few times.

**Comment by athul-mindtickle** [comment]
@seizethedave Any update on this?

**Comment by seizethedave** [comment]
> @seizethedave Any update on this?

@athul-mindtickle I haven’t found time to get to it yet with all the quarantine business. I hope in the next ~month i’ll be able to address it.

**Comment by seizethedave** [comment]
@athul-mindtickle I've just released 4.7.0 that will retry failed requests. Please give it a spin!
